### PR TITLE
feat: Imprement jwt-auth header auth method

### PR DESCRIPTION
**Pull request by Nerahikada**
**Description**
This change is based on #2474. See original PR for detailed description.

To configure the filebrowser for Cloudflare Zero Trust, run the following command:
```bash
./filebrowser config set \
  --auth.method=jwt-header \
  --auth.jwt-header.header=Cf-Access-Jwt-Assertion \
  --auth.jwt-header.aud=<your application AUD> \
  --auth.jwt-header.iss=https://<your team name>.cloudflareaccess.com \
  --auth.jwt-header.certsurl=https://<your team name>.cloudflareaccess.com/cdn-cgi/access/certs \
  --auth.jwt-header.usernameClaim=email
```

:rotating_light: Before submitting your PR, please read [community](https://github.com/filebrowser/community), and indicate which issues (in any of the repos) are either fixed or closed by this PR. See [GitHub Help: Closing issues using keywords](https://help.github.com/articles/closing-issues-via-commit-messages/).

- [x] DO make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] DO make sure you are making a pull request against the **master branch** (left side). Also you should start *your branch* off *our master*.
- [x] DO make sure that File Browser can be successfully built. See [builds](https://github.com/filebrowser/community/blob/master/builds.md) and [development](https://github.com/filebrowser/community/blob/master/development.md).
- [x] DO make sure that related issues are opened in other repositories. I.e., the frontend, caddy plugins or the web page need to be updated accordingly.
- [x] AVOID breaking the continuous integration build.

**Further comments**

### CODE REVIEW

**Comment by Nerahikada** [comment]
bump

File: auth/jwt.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	Header        string `json:"header"`
+	remoteKeySet  *oidc.RemoteKeySet
+	init          sync.Once
+}
+
+// Auth authenticates the user via a JWT token in an HTTP header.
+func (a *JWTAuth) Auth(r *http.Request, usr users.Store, stg *settings.Settings, srv *settings.Server) (*users.User, error) {
+	a.init.Do(func() {
+		a.remoteKeySet = oidc.NewRemoteKeySet(context.Background(), a.CertsURL)
+	})
```
Comment (o1egl) [inline auth/jwt.go:35]: I think it's better to do it in the constructor func
```go
func NewJWTAuth(...){
...
}
```

File: auth/jwt.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	if accessJWT == "" {
+		return nil, os.ErrPermission
+	}
+
+	// The Application Audience (AUD) tag for your application
+	config := &oidc.Config{
+		ClientID: a.Aud,
+	}
+
+	verifier := oidc.NewVerifier(a.Iss, a.remoteKeySet, config)
```
Comment (o1egl) [inline auth/jwt.go:47]: Looks like a good candidate to be moved into the constructor

File: auth/jwt.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+// Auth authenticates the user via a JWT token in an HTTP header.
+func (a *JWTAuth) Auth(r *http.Request, usr users.Store, stg *settings.Settings, srv *settings.Server) (*users.User, error) {
+	a.init.Do(func() {
+		a.remoteKeySet = oidc.NewRemoteKeySet(context.Background(), a.CertsURL)
+	})
+
+	accessJWT := r.Header.Get(a.Header)
+	if accessJWT == "" {
+		return nil, os.ErrPermission
```
Comment (o1egl) [inline auth/jwt.go:39]: It would be nice to log an error here

File: auth/jwt.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	// The Application Audience (AUD) tag for your application
+	config := &oidc.Config{
+		ClientID: a.Aud,
+	}
+
+	verifier := oidc.NewVerifier(a.Iss, a.remoteKeySet, config)
+
+	token, err := verifier.Verify(r.Context(), accessJWT)
+	if err != nil {
+		return nil, os.ErrPermission
```
Comment (o1egl) [inline auth/jwt.go:51]: Please log the original error

File: auth/jwt.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	payload := map[string]any{}
+	err = token.Claims(&payload)
+	if err != nil {
+		return nil, os.ErrPermission
```
Comment (o1egl) [inline auth/jwt.go:57]: Please log the original error

File: auth/jwt.go [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	user, err := usr.Get(srv.Root, payload[a.UsernameClaim])
+	if nerrors.Is(err, errors.ErrNotExist) {
+		return nil, os.ErrPermission
```
Comment (o1egl) [inline auth/jwt.go:62]: Please log the original error

File: auth/jwt.go [hunk 7]
```diff
@@ -0,0 +1,71 @@
+package auth
+
+import (
+	"context"
+	nerrors "errors"
```
Comment (o1egl) [inline auth/jwt.go]: I'd instead give an alias to the filebrowser errors
```
fbErrors "github.com/filebrowser/filebrowser/v2/errors"
```

Comment (Nerahikada) [reply auth/jwt.go]: fixed by d19962b70a41be6219f88c99a46e1a97a6ac1fb7

**Comment by Nerahikada** [comment]
stale

**Comment by Nerahikada** [comment]
can someone please take over?

**Pull request closed without merging**